BugsnagPerformance/Assets/BugsnagPerformance/Scripts/Public/BugsnagPerformance.cs

```diff
@@ -27,11 +27,22 @@ public class BugsnagPerformance
         private AppStartHandler _appStartHandler;
         private PersistentState _persistentState;
         private PValueUpdater _pValueUpdater;
-        private static List<Span> _potentiallyOpenSpans = new List<Span>();
+        private static List<WeakReference<Span>> _potentiallyOpenSpans = new List<WeakReference<Span>>();
         private Func<BugsnagNetworkRequestInfo, BugsnagNetworkRequestInfo> _networkRequestCallback;
+
+        private Dictionary<WeakReference<BugsnagUnityWebRequest>, Span> _networkSpans = new Dictionary<WeakReference<BugsnagUnityWebRequest>, Span>();
```

Comment:
So if a `WeakReference` key gets invalidated, the associated span also gets dropped? (just want to confirm that this is the behaviour of dictionaries with`WeakReference` keys, rather than the dictionary keeping the invalidated weak reference as a lookup to the still-hanging-around span)

Reply:
i have added a method that checks for collected web requests whenever a new request is sent or a request is completed or aborted. This will also then clean up the attached span